orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -24,6 +31,260 @@ func TestOrderbookUsecaseTestSuite(t *testing.T) {
 	suite.Run(t, new(OrderbookUsecaseTestSuite))
 }
 
+func (s *OrderbookUsecaseTestSuite) TestProcessPool() {
+	testCases := []struct {
+		name          string
+		pool          sqsdomain.PoolI
+		setupMocks    func(usecase *orderbookusecase.OrderbookUseCaseImpl, client *mocks.OrderbookGRPCClientMock, repository *mocks.OrderbookRepositoryMock)
+		expectedError string
+	}{
+		{
+			name:          "pool is nil",
+			pool:          nil,
+			expectedError: "pool is nil when processing order book",
```

Comment:
Let's also import the actual errors rather than hardcoding strings.

This is so that if we need to change an error message, we only need to do it in one place

Reply:
This is actually the identical situation as described in previous PR [comment](https://github.com/osmosis-labs/sqs/pull/492#discussion_r1738497416). If you would like still to proceed with this idea can we descope it from this task?

Reply:
Per the comment in [another PR](https://github.com/osmosis-labs/sqs/pull/492#discussion_r1740168398), I would like to suggest considering completing it in this PR so that we avoid accumulating tech debt.

IMO, this should be under 20 minutes to change/refactor but will yield the benefit of not having to switch context to come back to in the future and/or accumulate tech debt that is already exceeding the comfortable levels.

Will leave the final call to you though!